plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java

```diff
@@ -314,13 +322,33 @@ void deleteBlobs(Collection<String> blobNames) throws IOException {
             return;
         }
         final List<BlobId> blobIdsToDelete = blobNames.stream().map(blob -> BlobId.of(bucketName, blob)).collect(Collectors.toList());
-        final List<Boolean> deletedStatuses = SocketAccess.doPrivilegedIOException(() -> client().delete(blobIdsToDelete));
-        assert blobIdsToDelete.size() == deletedStatuses.size();
+        final List<Boolean> deletedStatuses = new ArrayList<>();
+        final int deleteCount = blobIdsToDelete.size();
+        SocketAccess.doPrivilegedIOException(() -> {
```

Comment:
I find the way you do it with partitions for AWS in https://github.com/elastic/elasticsearch/pull/40322 easier to read and follow, can you make these implementations consistent?
We can think about refactoring this part of the code into AbstractBlobContainer

Reply:
Not really IMO, things work a little differently here, because we collect the `Boolean` list to track failures instead of catching exceptions. I think keeping it concise with the `List`s here makes it a lot easier to understand that the ordering in the responses list and the blob id list must be the same?